Makefile

```diff
@@ -9,19 +9,17 @@ all: nvim
 
 CMAKE_PRG ?= $(shell (command -v cmake3 || echo cmake))
 CMAKE_BUILD_TYPE ?= Debug
-CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE)
-# Extra CMake flags which extend the default set
-CMAKE_EXTRA_FLAGS ?=
+CMAKE_FLAGS := -DCMAKE_BUILD_TYPE=$(CMAKE_BUILD_TYPE) $(CMAKE_FLAGS)
 
 # CMAKE_INSTALL_PREFIX
-#   - May be passed directly or as part of CMAKE_EXTRA_FLAGS.
+#   - May be passed directly or as part of CMAKE_FLAGS.
 #   - `checkprefix` target checks that it matches the CMake-cached value. #9615
-ifneq (,$(CMAKE_INSTALL_PREFIX)$(CMAKE_EXTRA_FLAGS))
-CMAKE_INSTALL_PREFIX := $(shell echo $(CMAKE_EXTRA_FLAGS) | 2>/dev/null \
+ifneq (,$(CMAKE_INSTALL_PREFIX)$(CMAKE_FLAGS))
```

Comment:
this check will always be true now ;_;

That's why this is a bit tricky. Though I like the direction of this PR

Reply:
Ok, so what did this actually do? The body of the condition gets `CMAKE_INSTALL_PREFIX` from `CMAKE_EXTRA_FLAGS`, ok.

But the condition just says that either `CMAKE_INSTALL_PREFIX` or `CMAKE_EXTRA_FLAGS` is nonempty. So if you set `CMAKE_INSTALL_PREFIX`, and then have it in `CMAKE_EXTRA_FLAGS`, the latter overrides, right?  And if you set `CMAKE_INSTALL_PREFIX`, but `CMAKE_EXTRA_FLAGS` is empty, then this results in an empty local `CMAKE_INSTALL_PREFIX`.

That's intended? I'd have guessed the priorities are explizit > part of `CMAKE_EXTRA_FLAGS` > part of `CMAKE_FLAGS`, but changing it might break stuff. As far as I can see, setting `CMAKE_INSTALL_PREFIX` as part of `CMAKE_FLAGS` did not have any effect, so it shouldn't now?

Confusing, to say the least. Viml really gives Makefiles a run for their money :P Please advice on the intended behavior.